### PR TITLE
jenkins e2e aws: add federation e2e job

### PR DESCRIPTION
**Pull request by colhom**
Depends on https://github.com/kubernetes/kubernetes/pull/31908
\cc @quinton-hoole @madhusudancs

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/577)

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml [hunk 1]
```diff
@@ -70,6 +70,22 @@
...
+              export PROJECT="k8s-jkns-e2e-aws"
+              export FEDERATION=true
+              export E2E_ZONES="us-west-1a us-west-2a us-east-1a"
+              export KUBERNETES_PROVIDER=aws
+              export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-aws"
+              export KUBE_AWS_ZONE=us-west-1a
+              export NETWORK_PROVIDER="kubenet"
+              export KUBE_NODE_OS_DISTRIBUTION="debian"
+              export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Federation\]"
+
```
Comment (madhusudancs) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml]: Looks like unnecessary whitespace,

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml [hunk 2]
```diff
@@ -70,6 +70,22 @@
...
+            description: 'Run federation e2e tests on AWS using latest successful kubernetes-federation build.'
+            cron-string: '{sq-cron-string}'
+            trigger-job: 'kubernetes-federation-build'
+            job-env: |
+              export E2E_NAME="e2e-aws-federation-master"
+              export PROJECT="k8s-jkns-e2e-aws"
+              export FEDERATION=true
+              export E2E_ZONES="us-west-1a us-west-2a us-east-1a"
+              export KUBERNETES_PROVIDER=aws
+              export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-aws"
```
Comment (madhusudancs) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml]: You also need `FEDERATION_NAME`, `DNS_ZONE_NAME` and `FEDERATION_DOMAIN_NAME` variables. See the config for GCE.

**Comment by spxtr** [comment]
ok to test
cc @zmerlynn

**Comment by spxtr** [comment]
ok to test

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml (hunk 2 above)
Comment (colhom) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml]: I was only able to find `DNS_ZONE_NAME` and `FEDERATION_DOMAIN_MAP`

**Comment by zmerlynn** [comment]
Do you have plans for doing federation anywhere *but* bash? If not, we need to talk, because `kube-up.sh` is long term dead for AWS, and I'm planning to rip the bash based deployments out of the test infrastructure next release.

**Comment by zmerlynn** [comment]
cc @kubernetes/sig-aws

**Comment by colhom** [comment]
@madhusudancs ping

**Comment by madhusudancs** [comment]
@colhom thanks for pinging.

This PR needs a rebase, could you rebase it?

**Comment by colhom** [comment]
@madhusudancs rebased

**Comment by madhusudancs** [comment]
@ixdy @spxtr I don't understand the failure in travis. Should I ignore it and just merge this PR?

**Comment by spxtr** [comment]
> Jenkins Job kubernetes-e2e-aws-federation does not have a matching testgrid test

You need to add an entry to `testgrid/config`.

**Comment by colhom** [comment]
@spxtr will do

@quinton-hoole you mentioned want to start a new tab called `google-federation-aws` with our new job in it? Otherwise I would be inclined to add this job to the existing `google-federation` tab.

**Comment by quinton-hoole** [comment]
@colhom I don't feel strongly about the organization of the tabs/pages in the UI. It just occurred to me that for every federation e2e test running on GCE (e.g. normal, soak, slow, disruptive, scale etc over time), there's likely to be an equivalent one on GKE, AWS etc.  So it might make sense to have a separate page per platform.  But we could equally well to that reorganization later if it becomes necessary. Up to you.

**Comment by colhom** [comment]
So, interestingly enough fetching a [tarball](https://storage.googleapis.com/kubernetes-federation-release/ci/v1.5.0-alpha.2.415+3fe2de05accff5/kubernetes.tar.gz) from a very [recent successful build](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gce-federation/5455?log) works fine- which leads me to believe it's not the ACLs on the bucket.

Investigating further.

**Comment by colhom** [comment]
Whoops, sorry wrong tab!

**Comment by colhom** [comment]
@madhusudancs this should be ready to go.

\cc @spxtr

**Comment by madhusudancs** [comment]
@colhom LGTM.

@spxtr (for the sanity check), can you please take a look?

**Comment by spxtr** [comment]
Please fix the merge conflicts and assign a reviewer.

**Comment by fejta** [comment]
Please reopen after rebasing.
